project/src/js/ui/round/view/roundView.js

```diff
@@ -375,20 +375,30 @@ function renderGameEndedActions(ctrl) {
     m('strong', result), m('br')
   ];
   resultDom.push(m('em.resultStatus', status));
-  const buttons = ctrl.data.player.spectator ? [
-    button.shareLink(ctrl),
-    button.sharePGN(ctrl),
-    button.analysisBoard(ctrl),
-    ctrl.data.tv ? tvChannelSelector(ctrl) : null
-  ] : [
-    button.shareLink(ctrl),
-    button.sharePGN(ctrl),
-    button.newOpponent(ctrl),
-    button.answerOpponentRematch(ctrl),
-    button.cancelRematch(ctrl),
-    button.rematch(ctrl),
-    button.analysisBoard(ctrl)
-  ];
+  let buttons = null;
+  if (ctrl.data.tournament) {
```

Comment:
I'm a bit confused about how the current control flow for the `end:` handler within round's `socketHandler` works. It seems to prevent the popup with the game action from being created if the user is a spectator. Therefore I'm not sure how the button code for the spectators would ever be relevant. I'm sure that I'm missing something, but for the time being I've changed:

`if (!ctrl.data.player.spectator) {`

to read:
`if (ctrl.data.tournament || !ctrl.data.player.spectator) {`

so as not to affect things outside of tournaments.